.github/workflows/testing.yaml

```diff
@@ -0,0 +1,30 @@
+name: Tests and coverage
+
+on:
+  - push
+
+jobs:
+  testing:
+    strategy:
+      matrix:
+        python-version: [ "3.11", "3.12", "3.13" ]
+    name: Python Testing and Coverage
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+      - run: |
+          make install
+          make check
+          make test
```

Comment:
Maybe is good if you make it dependent: ```  make install &&   make check  &&  make test ``` so if one of the steps fails will stop the rest.

Reply:
I thought they could continue! That is good to know!